pkg/helm/metrics/metrics.go

```diff
@@ -0,0 +1,83 @@
+package metrics
+
+import (
+	"log"
+	"os"
+
+	"helm.sh/helm/v3/pkg/action"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest"
+	k8smetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	helmChartReleaseHealthStatus = k8smetrics.NewGaugeVec(
+		&k8smetrics.GaugeOpts{
+			Name: "helm_chart_release_health_status",
+			Help: "Health of the Helm release",
+		},
+		[]string{"releaseName", "chartName", "chartVersion"},
+	)
+)
+
+func init() {
+	legacyregistry.MustRegister(helmChartReleaseHealthStatus)
+}
+
+func HandleHelmChartReleaseHealthStatus() {
+	defer recoverMetricPanic()
+
+	if actionConfig, err := getActionConfig(); err != nil {
+		klog.Errorf("metric helm_chart_release_health_status unhandled: %v", err)
+		return
+	} else {
+		listAction := action.NewList(actionConfig)
+		releases, err := listAction.Run()
+		if err != nil {
+			klog.Errorf("metric helm_chart_release_health_status unhandled: %v", err)
+			return
+		}
+
+		for _, release := range releases {
+			if status := release.Info.Status.String(); status == "deployed" {
+				klog.V(4).Infof("metric helm_chart_release_health_status 1: %s %s %s", release.Name, release.Chart.Metadata.Name, release.Chart.Metadata.Version)
+				helmChartReleaseHealthStatus.WithLabelValues(release.Name, release.Chart.Metadata.Name, release.Chart.Metadata.Version).Set(1)
```

Comment:
the only diff between this and the line below is the 0 or 1. I would do healthStatus := 1 then change to 0 if status unknown or failed. Then make a single call to klog and to helmChartReleaseHealthStatus passing healthStatus